src/toil/test/jobStores/jobStoreTest.py

```diff
@@ -277,21 +277,24 @@ def callback(f2):
             stats = set()
             self.assertEquals(0, master.readStatsAndLogging(callback))
             self.assertEquals(set(), stats)
-            master.writeStatsAndLogging('1')
+            worker.writeStatsAndLogging('1')
             self.assertEquals(1, master.readStatsAndLogging(callback))
             self.assertEquals({'1'}, stats)
             self.assertEquals(0, master.readStatsAndLogging(callback))
-            master.writeStatsAndLogging('1')
-            master.writeStatsAndLogging('2')
+            worker.writeStatsAndLogging('1')
+            worker.writeStatsAndLogging('2')
             stats = set()
             self.assertEquals(2, master.readStatsAndLogging(callback))
             self.assertEquals({'1', '2'}, stats)
             largeLogEntry = os.urandom(self._largeLogEntrySize())
             stats = set()
-            master.writeStatsAndLogging(largeLogEntry)
+            worker.writeStatsAndLogging(largeLogEntry)
             self.assertEquals(1, master.readStatsAndLogging(callback))
             self.assertEquals({largeLogEntry}, stats)
 
+            # test the readAll parameter
+            self.assertEqual(4, master.readStatsAndLogging(callback, readAll=True))
```

Comment:
Due to stats simplifications, I can no longer guarantee we will always get the specified number of stats files in any of these assertions. So far I haven't seen any failures and when/if I do I can increase the grace period at the end of readStatsAndLogging to a higher value. Another alternative would be to just remove these assertions altogether but I think we should keep them in for now